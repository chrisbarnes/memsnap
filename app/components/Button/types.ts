export interface IButtonProps {
  children: React.ReactNode;
  onClick?: () => void;
  type?: "button" | "submit" | "reset";
  variant?: "primary" | "secondary";
}
