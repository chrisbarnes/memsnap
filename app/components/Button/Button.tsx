import type { IButtonProps } from "./types";

export const Button = ({
  children,
  onClick,
  type = "button",
  variant = "primary",
}: IButtonProps) => {
  let classNames = "";

  if (variant === "primary") {
    classNames =
      "text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 me-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800";
  }

  if (variant === "secondary") {
    classNames =
      "focus:outline-none text-white bg-green-700 hover:bg-green-800 focus:ring-4 focus:ring-green-300 font-medium rounded-lg text-sm px-5 py-2.5 me-2 mb-2 dark:bg-green-600 dark:hover:bg-green-700 dark:focus:ring-green-800";
  }

  return (
    <button type={type} className={classNames} onClick={onClick}>
      {children}
    </button>
  );
};
