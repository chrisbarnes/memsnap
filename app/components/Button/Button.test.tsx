// tests for Button.tsx
import { render } from "@testing-library/react";
import { Button } from "./Button";

describe("Button", () => {
  it("renders a primary button correctly", () => {
    const { container } = render(<Button variant="primary">Click Me!</Button>);
    expect(container).toMatchSnapshot();
  });

  it("renders a secondary button correctly", () => {
    const { container } = render(
      <Button variant="secondary">Click Me!</Button>
    );
    expect(container).toMatchSnapshot();
  });
});
