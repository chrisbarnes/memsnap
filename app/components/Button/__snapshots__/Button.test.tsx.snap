// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders a primary button correctly 1`] = `
<div>
  <button
    class="text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 font-medium rounded-lg text-sm px-5 py-2.5 me-2 mb-2 dark:bg-blue-600 dark:hover:bg-blue-700 focus:outline-none dark:focus:ring-blue-800"
    type="button"
  >
    Click Me!
  </button>
</div>
`;

exports[`Button renders a secondary button correctly 1`] = `
<div>
  <button
    class="focus:outline-none text-white bg-green-700 hover:bg-green-800 focus:ring-4 focus:ring-green-300 font-medium rounded-lg text-sm px-5 py-2.5 me-2 mb-2 dark:bg-green-600 dark:hover:bg-green-700 dark:focus:ring-green-800"
    type="button"
  >
    Click Me!
  </button>
</div>
`;
